src/Features/Core/Portable/Diagnostics/BuildOnlyDiagnosticsService.cs

```diff
@@ -61,18 +56,12 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
             }
         }
 
-        public void AddBuildOnlyDiagnostics(Solution solution, ProjectId? projectId, DocumentId? documentId, ImmutableArray<DiagnosticData> diagnostics)
+        public void AddBuildOnlyDiagnostics(DocumentId documentId, ImmutableArray<DiagnosticData> diagnostics)
```

Comment:
since nothing was reading the project side, we only need to call this for teh document side of things.